File: sdk/php/src/Resources/NetworkServices.php
<?php

namespace HiTechCloud\SDK\Resources;

class NetworkServices extends BaseResource
{
    /**
     * Get IPs
     */
    public function getIps(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/ip');
    }

    /**
     * Get rDNS
     */
    public function getRdns(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/rdns');
    }

    /**
     * Set rDNS
     */
    public function setRdns(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/rdns', $data);
    }

}

File: sdk/csharp/Resources/NetworkServices.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class NetworkServices : BaseResource
{
    public NetworkServices(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> IpsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/ip", id));
    }

    public async Task<Dictionary<string, object?>> RdnsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/rdns", id));
    }

    public async Task<Dictionary<string, object?>> SetRdnsAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/rdns", id), data);
    }
}
